src/main/java/seedu/address/logic/commands/ModeCommand.java

```diff
@@ -47,7 +54,12 @@ public ModeCommand(String newGameModeName, String newGameDifficultyName) {
     public CommandResult executePrimitive(Model model, CommandHistory history) {
         requireNonNull(model);
 
-        if (newGameDifficultyName == null) {
+        if (newGameModeName == null && newGameDifficultyName == null) {
+            GameMode gameMode = model.getTaskManager().getGameManager().getGameMode();
```

Comment:
LoD violation?

Reply:
I don't think getTaskManager returns a copy.

Reply:
Good catch, updated! now calls getGameMode in TaskManager, which then returns a copy of the game mode.